Palette.js

```diff
@@ -9,9 +9,19 @@ class Palette {
       new Color(),
     ];
   }
+  
   refreshColors() {
     for (var i = 0; i < this.colors.length; i++) {
       this.colors[i].refreshHexCode()
     }
   }
+
+  toggleColorLock(boxId) {
+    for (var i = 0; i < this.colors.length; i++) {
+      if (boxId === this.colors[i].id) {
+        this.colors[i].isLocked = !this.colors[i].isLocked;
+        return i
+      }
+    }
+  }
```

Comment:
Made toggling the lock value of the clicked color a method of the Palette Class since the spec said this needed be a part of the class.